tests/test_refObjLoader.py

```diff
@@ -0,0 +1,294 @@
+        self.assertLess(np.max(dist), 30.0)
+
+        # Check that all the objects from the two catalogs are here.
+        dist = sphdist(180.0, 0.0, self.skyCatalog['ra_icrs'], self.skyCatalog['dec_icrs'])
+        inside, = (dist < 30.0).nonzero()
+        self.assertEqual(len(cat), len(inside))
+
+        self.assertTrue(cat.isContiguous())
+        self.assertEqual(len(np.unique(cat['id'])), len(cat))
+        # A default-loaded sky circle should not have centroids
+        self.assertNotIn('centroid_x', cat.schema)
+        self.assertNotIn('centroid_y', cat.schema)
+        self.assertNotIn('hasCentroid', cat.schema)
+
+    def test_loadPixelBox(self):
+        """Test the loadPixelBox routine."""
+        # This will create a box 50 degrees on a side.
+        loaderConfig = ReferenceObjectLoader.ConfigClass()
+        loaderConfig.pixelMargin = 0
+        loader = ReferenceObjectLoader([dataRef.dataId for dataRef in self.datasetRefs],
+                                       self.handles,
+                                       config=loaderConfig)
+        bbox = lsst.geom.Box2I(corner=lsst.geom.Point2I(0, 0), dimensions=lsst.geom.Extent2I(1000, 1000))
+        crpix = lsst.geom.Point2D(500, 500)
+        crval = lsst.geom.SpherePoint(180.0*lsst.geom.degrees, 0.0*lsst.geom.degrees)
+        cdMatrix = afwGeom.makeCdMatrix(scale=0.05*lsst.geom.degrees)
+        wcs = afwGeom.makeSkyWcs(crpix=crpix, crval=crval, cdMatrix=cdMatrix)
+
+        cat = loader.loadPixelBox(bbox, wcs, 'a', bboxToSpherePadding=0).refCat
+
+        # This is a sanity check on the ranges; the exact selection depends
+        # on cos(dec) and the tangent-plane projection.
+        self.assertLess(np.max(np.rad2deg(cat['coord_ra'])), 180.0 + 25.0)
+        self.assertGreater(np.max(np.rad2deg(cat['coord_ra'])), 180.0 - 25.0)
+        self.assertLess(np.max(np.rad2deg(cat['coord_dec'])), 25.0)
+        self.assertGreater(np.min(np.rad2deg(cat['coord_dec'])), -25.0)
+
+        # The following is to ensure the reference catalog coords are
+        # getting corrected for proper motion when an epoch is provided.
+        # Use an extreme epoch so that differences in corrected coords
+        # will be significant.  Note that this simply tests that the coords
+        # do indeed change when the epoch is passed.  It makes no attempt
+        # at assessing the correctness of the change.  This is left to the
+        # explicit testProperMotion() test below.
+        catWithEpoch = loader.loadPixelBox(
+            bbox,
+            wcs,
+            'a',
+            bboxToSpherePadding=0,
+            epoch=astropy.time.Time(30000, format='mjd', scale='tai')).refCat
+
+        self.assertFloatsNotEqual(cat['coord_ra'], catWithEpoch['coord_ra'], rtol=1.0e-4)
+        self.assertFloatsNotEqual(cat['coord_dec'], catWithEpoch['coord_dec'], rtol=1.0e-4)
+
+    def test_filterMap(self):
```

Comment:
I think I'd rather this test didn't exist, and we just relied on the ones in `test_loadReferenceObjects.py`, but I don't know if this one provides any specific coverage that that doesn't.

Reply:
I'm copying over the tests that were there before in gen2-only land.  But the slight difference is that this is testing the return values of `loadSkyCircle` which the tests in `test_loadReferenceObjects.py` are not (because it has a dummy `loadSkyCircle`).  The second test (that the values are equal) is I guess redundant with the schema mapping tests, but it's not expensive.

Reply:
In particular, this test hits https://github.com/lsst/meas_algorithms/blob/f36ae6ace101d9874c6f103cd9182f19e188b404/python/lsst/meas/algorithms/loadReferenceObjects.py#L489